### PR TITLE
Wrote user DAO and test

### PR DESCRIPTION
**Pull request by justin-obrien-smoothstack**
This includes #3, and #13, so after those get merged there will be fewer changes to review here.

### CODE REVIEW

File: src/test/java/com/ss/training/utopia/counter/dao/UserDaoTests.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	private UserDao userDao;
+
+	@Test
+	public void findByUsernameTest() {
+		String thisUsername = "ThisUsername", otherUsername = "OtherUsername";
+		User savedUser = new User(null, thisUsername, null, null, null),
+				otherUser = new User(null, otherUsername, null, null, null), foundUser;
+		testEntityManager.persist(savedUser);
+		testEntityManager.persist(otherUser);
+		testEntityManager.flush();
```
Comment (trevorhuis-smoothstack) [inline src/test/java/com/ss/training/utopia/counter/dao/UserDaoTests.java:31]: What are these three commands with the testEntityManager doing?

**Review by trevorhuis-smoothstack — APPROVED** [review]
Looks like a good start to testing, What are you using the H2 dependency for?

File: src/test/java/com/ss/training/utopia/counter/dao/UserDaoTests.java (hunk 1 above)
Comment (justin-obrien-smoothstack) [reply src/test/java/com/ss/training/utopia/counter/dao/UserDaoTests.java:31]: `persist` designates entities to be saved in H2, and `flush` more or less commits those changes.

**Comment by justin-obrien-smoothstack** [comment]
It provides an in-memory database for testing (as we discussed in the meeting just now).

**Pull request merged**